### ISSUE TITLE
TypeError index.js:83

### ISSUE THREAD

**Issue by chisleu** [codemanki/cloudscraper#9]
```
  makeRequest(options, function(error, response, body) {
    var validationError;
    var stringBody = body.toString('utf8');

    if (validationError = checkForErrors(error, stringBody)) {
      return callback(validationError, body, response);
    }

    // If body contains specified string, solve challenge
    if (stringBody.indexOf('a = document.getElementById(\'jschl-answer\');') !== -1) {
      setTimeout(function() {
        return solveChallenge(response, stringBody, options, callback);
      }, Timeout);
    } else {
      // All is good
      giveResults(options, error, response, body, callback);
    }
  });
}

function checkForErrors(error, body) {
  var match;

  // Pure request error (bad connection, wrong url, etc)
  if(error) {
    return { errorType: 0, error: error };
  }

  // Finding captcha
  if (body.indexOf('why_captcha') !== -1 || /recaptcha/i.test(body)) {
    return { errorType: 1 };
  }

  // trying to find '<span class="cf-error-code">1006</span>'
  match = body.match(/<\w+\s+class="cf-error-code">(.*)<\/\w+>/i);

  if (match) {
    return { errorType: 2, error: parseInt(match[1]) };
  }

  return false;
}
```
body.toString() is called unsafely. Some request errors cause body to be undefined.

**Comment by codemanki** [comment]
@chisleu thanks for reporting this. Do you have an example scenario or actual error response when this happens? I would like to implement a test case for this issue.

**Comment by chisleu** [comment]
I want to say it is malformed DNS responses, but sadly I can not share the hostnames in question.

**Comment by codemanki** [comment]
Could you maybe just share the response without hostnames and any other private information?

**Comment by chisleu** [comment]
I'll try tomorrow.

**Comment by chisleu** [comment]
All of the errors were transient and cleared up overnight. Because it throws an uncatchable error, I'm setting up a handler for that, but since my process is highly parallel, I can't pull the hostname unless I'm actively monitoring when an error occurs.

Either way, it seems like a reasonably easy check to see if(body) before you toString().

**Comment by codemanki** [comment]
@chisleu please take a look https://github.com/codemanki/cloudscraper/pull/10

**Comment by chisleu** [comment]
That looks decent, but I got another site to trigger the issue just now. 

Response: undefined
Body: undefined

I didn't think to print anything else.

I can apply that patch tomorrow and let it run to see if I get a different crash. I haven't traced back the callbacks to see if it breaks later in the chain on error.

**Comment by chisleu** [comment]
I killed it after it crashed on a new site and replaced the cloudscraper with that patch #10 , and it didn't crash when it re-ran, so I think it is good to go.

I will report if something changes, but thumbs up here.

**Comment by codemanki** [comment]
@chisleu cool. thanks. releasing the pr right nau

**Comment by codemanki** [comment]
Fix is available in 1.2.1
